mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt

```diff
@@ -1,79 +1,90 @@
+        BadgeType.Admin
+      } else if (tournament.isParticipant) {
+        BadgeType.Participant
+      } else {
+        BadgeType.Join
+      }
+  // TODO: Change to tournament.status when added.
+  override val status: Status = Status.Done
+}
 
 /**
  * An implementation of the [ContestScreenState] that performs a given profile's [ContestInfo]
  * requests.
+ *
+ * @param currentUser the current [AuthenticatedUser] of the application.
+ * @param tournamentFacade the [TournamentFacade] used to perform some requests.
+ * @param scope the [CoroutineScope] on which requests are performed.
  */
-class TournamentScreenState : ContestScreenState<ContestInfoAdapter> {
+class TournamentScreenState(
+    private val currentUser: AuthenticatedUser,
+    private val tournamentFacade: TournamentFacade,
+    private val scope: CoroutineScope,
+) : ContestScreenState<TournamentAdapter> {
+
+  override var contests by mutableStateOf(emptyList<TournamentAdapter>())
+    private set
 
-  // TODO : Fill this in with some actual data.
-  override val contests =
-      listOf(
-          createContest("1", "EPFL Grand Prix", Status.Ongoing(1.days), BadgeType.Admin),
-          createContest("2", "Pawn Party", Status.Done, BadgeType.Participant),
-          createContest("3", "Never gonna chess", Status.Ongoing(3.hours), BadgeType.Join),
-      )
+  init {
+    scope.launch {
+      tournamentFacade.getTournaments(currentUser).collect { list ->
+        contests = list.map { TournamentAdapter(it, currentUser) }
+      }
+    }
+  }
 
   override fun onNewContestClick() = Unit
-  override fun onContestClick(contest: ContestInfoAdapter) = Unit
+  override fun onContestClick(contest: TournamentAdapter) = Unit
   override fun onFilterClick() = Unit
 }
 
-/**
- * TODO : Remove this.
- *
- * Creates a contest with some fake data.
- *
- * @param uid the id of the contest.
- * @param name the title of the contest.
- * @param status the [Status] for the contest.
- * @param personStatus the badge to display.
- */
-private fun createContest(
-    uid: String,
-    name: String,
-    status: Status,
-    personStatus: BadgeType?
-): ContestInfoAdapter {
-  return ContestInfoAdapter(uid, name, personStatus, status)
-}
-
 /**
  * A stateful composable to view the list of tournaments completed and ongoing.
  *
+ * @param currentUser the current [AuthenticatedUser] of the application.
  * @param modifier the [Modifier] for this composable.
  * @param contentPadding the [PaddingValues] for this composable.
  */
 @Composable
 fun StatefulTournamentScreen(
+    currentUser: AuthenticatedUser,
     modifier: Modifier = Modifier,
     contentPadding: PaddingValues = PaddingValues(),
 ) {
-  ContestScreen(TournamentScreenState(), modifier, key = { it.uid }, contentPadding)
+  val tournamentFacade = LocalTournamentFacade.current
+  val scope = rememberCoroutineScope()
+
+  ContestScreen(
+      TournamentScreenState(currentUser, tournamentFacade, scope),
```

Comment:
Missing a `remember(currentUser, tournamentFacade, scope) { ... }` ^^ ?

Reply:
Yes! Always forget! ^^
Fixed in 5d32a9fb42579af9f705b9889170f812091800b2